crowbar_framework/app/controllers/nodes_controller.rb

```diff
@@ -304,22 +304,38 @@ def status
   end
 
   def hit
-    action = params[:req]
     name = params[:name] || params[:id]
     machine = NodeObject.find_node_by_name name
-    if machine.nil?
-      render text: "Could not find node '#{name}'", status: 404 and return
-    else
-      case action
-      when "reinstall", "reset", "update", "delete"
-        machine.set_state(action)
-      when "reboot", "shutdown", "poweron", "powercycle", "poweroff", "identify", "allocate"
-        machine.send(action)
-      else
-        render text: "Invalid hit request '#{action}'", status: 500 and return
+
+    respond_to do |format|
+      format.json do
+        if machine.nil?
+          render json: {
+            error: I18n.t("nodes.hit.not_found", name: name)
+          }, status: 404
+        elsif machine.actions.include? params[:req].to_s
```

Comment:
Not if `params[:req]` equals to `nil` which can happen if it's called through a wrong route.